distribution/pull_v2.go

```diff
@@ -147,6 +149,10 @@ func (ld *v2LayerDescriptor) DiffID() (layer.DiffID, error) {
 	return ld.V2MetadataService.GetDiffID(ld.digest)
 }
 
+func (ld *v2LayerDescriptor) ForeignSource() *distribution.Descriptor {
+	return ld.foreignSrc
```

Comment:
As currently designed, only layers that got a descriptor from `docker load` or had external URLs from `docker pull` will store a descriptor.